Sources/Player/PlaybackEngine/Internal/Cache/DBCacheStorage.swift

```diff
@@ -0,0 +1,119 @@
+import Foundation
+import GRDB
+
+// MARK: - DBCacheStorage
+
+class DBCacheStorage {
+	private let dbQueue: DatabaseQueue
+
+	init(dbQueue: DatabaseQueue) {
+		self.dbQueue = dbQueue
+		try? initializedDatabase()
+	}
+
+	// MARK: - Calculate Total Size
+
+	func totalSize() throws -> Int {
+		try dbQueue.read { db in
+			try calculateTotalSize(db)
+		}
+	}
+
+	// MARK: - Prune to a Maximum Size
+
+	func pruneToSize(_ maxSize: Int) throws {
```

Comment:
The final usage is not in the PRs yet. It would mean making the `AVQueuePlayerWrapperLegacy` compatible with this new code which is a pain. So the integration is on hold until we cna get rid of the legacy player implementation. 

But the ideal will be that every time an item is added to the cache, this prunetoSize() will be called to enforce the maximum cache size.

But I will keep it in mind and think about the error handling a bit more.

Reply:
Got it! Yes, it will be great to have it in mind since there are so many throws in each PR and therefore many points of failure.